SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef SMALL_BETTER_LOGGER_H
+#define SMALL_BETTER_LOGGER_H
+
+// Used for writing to output stream
+#include <iostream>
+#include <fstream>
+
+// Used for formatting and creating the output string
+#include <sstream>
+#include <vector>
+#include <regex>
+
+namespace sbl
+{
+	// Logger declaration
+	class Logger;
+	using logger = Logger;
+
+	// Stream types to be used by a Logger instance
+	enum class STREAM_TYPE
+	{
+		STDOUT, STDERR, STDLOG, FILE
+	};
+	using stream_type = STREAM_TYPE;
+
+	// Logger class: The main part of the library, used to log messages to a stream (ex.: STDOUT, STDERR, file etc.)
+	class Logger
```

Comment:
This could come in the future as a whole new feature. You could weigh in the pros and cons of adding a static logger with Initialize and Finalize methods. Check out more deisgn patterns used for logger classes. 
[https://medium.com/@nempet/multiple-logging-sources-using-strategy-design-pattern-1b5877c75966](url)